### ISSUE TITLE
"systemctl edit" fails to create service templates

### ISSUE THREAD

**Issue by vgaetera** [systemd/systemd#32922]
### systemd version the issue has been seen with

255.6

### Used distribution

Fedora 40

### Linux kernel version used

kernel-6.8.9-300.fc40.x86_64

### CPU architectures issue was seen on

x86_64

### Component

systemctl

### Expected behaviour you didn't see

The behavior is expected to be similar to creating a normal service unit.

### Unexpected behaviour you saw

Nothing happens, the exit code indicates an error.

### Steps to reproduce the problem

```sh
sudo systemctl edit --force --full test@.service
```

**Comment by zootboy** [comment]
Ran into this on systemd 256 (256.9-1-arch). Here's the output with loglevel set to debug:

```
$ sudo SYSTEMD_LOG_LEVEL=debug systemctl edit --full --force thing@.service
Bus n/a: changing state UNSET → OPENING
sd-bus: starting bus by connecting to /run/systemd/private...
Bus n/a: changing state OPENING → AUTHENTICATING
Bus n/a: changing state AUTHENTICATING → RUNNING
Sent message type=method_call sender=n/a destination=org.freedesktop.systemd1 path=/org/freedesktop/systemd1 interface=org.freedesktop.systemd1.Manager member=GetUnitFileState cookie=1 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=error sender=org.freedesktop.systemd1 destination=n/a path=n/a interface=n/a member=n/a  cookie=1 reply_cookie=1 signature=s error-name=org.freedesktop.DBus.Error.FileNotFound error-message=No such file or directory
Failed to get UnitFileState for 'thing@.service': No such file or directory
Bus n/a: changing state RUNNING → CLOSED
```
 Returns 1.

Also tried a few variants of backslash-escaping and quoting, but no difference.

**Comment by yuwata** [comment]
Recently fixed by b58b00e4c33474505009c8118d6cfdf29a2c6cb1.
